tests/test_events.py

```diff
@@ -0,0 +1,751 @@
+                    'path': 'd'
+                }
+            ]
+        }
+
+    def test_list_of_files(self):
+        files = []
+        utils.list_of_files(self.tree, files)
+        assert_equal(['e', 'f', 'c', 'd'], files)
+
+    def tearDown(self):
+        pass
+
+
+class TestEventExists(OsfTestCase):
+    # Add all possible called events here to ensure that the Event class can
+    #  call them.
+    def setUp(self):
+        super(TestEventExists, self).setUp()
+        self.user = factories.UserFactory()
+        self.consolidate_auth = Auth(user=self.user)
+        self.node = factories.ProjectFactory(creator=self.user)
+
+    def test_get_file_updated(self):
+        # Event gets FileUpdated from file_updated
+        event = event_register['file_updated'](self.user, self.node, 'file_updated', payload=file_payload)
+        assert_is_instance(event, FileUpdated)
+
+    def test_get_file_added(self):
+        # Event gets FileAdded from file_added
+        event = event_register['file_added'](self.user, self.node, 'file_added', payload=file_payload)
+        assert_is_instance(event, FileAdded)
+
+    def test_get_file_removed(self):
+        # Event gets FileRemoved from file_removed
+        event = event_register['file_removed'](self.user, self.node, 'file_removed', payload=file_deleted_payload)
+        assert_is_instance(event, FileRemoved)
+
+    def test_get_folder_created(self):
+        # Event gets FolderCreated from folder_created
+        event = event_register['folder_created'](self.user, self.node, 'folder_created', payload=folder_created_payload)
+        assert_is_instance(event, FolderCreated)
+
+    def test_get_file_moved(self):
+        # Event gets AddonFileMoved from addon_file_moved
+        file_moved_payload = file_move_payload(self.node, self.node)
+        event = event_register['addon_file_moved'](self.user, self.node, 'addon_file_moved', payload=file_moved_payload)
+        assert_is_instance(event, AddonFileMoved)
+
+    def test_get_file_copied(self):
+        # Event gets AddonFileCopied from addon_file_copied
+        file_copied_payload = file_copy_payload(self.node, self.node)
+        event = event_register['addon_file_copied'](self.user, self.node, 'addon_file_copied', payload=file_copied_payload)
+        assert_is_instance(event, AddonFileCopied)
+
+    def tearDown(self):
+        pass
+
+
+class TestSignalEvent(OsfTestCase):
+    def setUp(self):
+        super(TestSignalEvent, self).setUp()
+        self.user = factories.UserFactory()
+        self.auth = Auth(user=self.user)
+        self.node = factories.ProjectFactory(creator=self.user)
+
+    # @mock.patch('website.notifications.events.files.FileAdded.perform')
+    def test_event_signal(self):  # , mock_perform):
+        signals.file_updated.send(
+            user=self.user, node=self.node, event_type='file_added', payload=file_payload
+        )
+        # assert_true(mock_perform.called)
+
+    def tearDown(self):
```

Comment:
Overriding this with `pass` will mean that the superclass's `tearDown` won't be executed. Remove this.